common/functions/src/scalars/comparisons/comparison.rs

```diff
@@ -131,10 +131,12 @@ impl<T: ComparisonImpl> ComparisonFunctionCreator<T> {
 
         let lhs_id = args[0].data_type_id();
```

Comment:
column is physical, so it's data_type is always physical.

For example:
A column with `timestamp` values, it's UInt64Column physically.

If we use `ColumnRef` to replace `ColumnsWithField`, we should keep DataFields inside the function.

Reply:
@sundy-li  Yes, I noticed it. The arguments types of a `Function` should be determined during instantiating, which will reduce many bugs.

But I found most of the functions are not fully specialized, then it's a hard work to decouple `DataField` with `Function`.

In new planner, we will perform strict type checking. Available type cast will be added, and the type errors will be raised during planning. Therefore I'm going to continue the work as soon as new planner is enabled by default.

Reply:
Ok.

> Available type cast will be added, and the type errors will be raised during planning.

Now we rejected the datatypes inside the function's creator fn (arg type validation is inside creator) and call `return_type` to have the return datatype